contracts/oracles/WstEthOracle.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: bsl-1.1
+
+pragma solidity 0.7.6;
+
+import "../helpers/SafeMath.sol";
+import "../helpers/ERC20Like.sol";
+import "../interfaces/IWstEthToken.sol";
+import "../interfaces/IStEthPriceFeed.sol";
+import "../interfaces/IOracleUsd.sol";
+import "../interfaces/IOracleRegistry.sol";
+import "../interfaces/IOracleEth.sol";
+import "../VaultParameters.sol";
+
+/**
+ * @title WstEthOracle
+ * @dev Wrapper to quote wstETH ERC20 token that represents the account's share of the total supply of stETH tokens. https://docs.lido.fi/contracts/wsteth/
+ **/
+
+contract WstEthOracle is IOracleUsd, Auth  {
+    using SafeMath for uint;
+
+    IOracleRegistry public immutable oracleRegistry;
+
+    uint immutable stEthDecimals;
+
+    address public stEthPriceFeed;
+
+    address public WETH;
+
+
+    event StEthPriceFeedChanged(address indexed implementation);
+
+    constructor(address _vaultParameters, address _oracleRegistry,  uint _stETHDecimals, address _stETHPriceFeed) Auth(_vaultParameters) {
+        require(_vaultParameters != address(0) && _oracleRegistry != address(0) && _stETHPriceFeed != address(0), "Unit Protocol: ZERO_ADDRESS");
+        oracleRegistry = IOracleRegistry(_oracleRegistry);
+        WETH = IOracleRegistry(_oracleRegistry).WETH();
+        require(WETH != address(0), "Unit Protocol: ZERO_ADDRESS");
+        stEthDecimals = _stETHDecimals;
+        stEthPriceFeed = _stETHPriceFeed;
+    }
+
+    function setStEthPriceFeed(address impl) external onlyManager {
+      require(impl != address(0), "Unit Protocol: ZERO_ADDRESS");
+      stEthPriceFeed = impl;
+      emit StEthPriceFeedChanged(impl);
+    }
+
+    function getDecimalsStEth() public view returns (uint) {
+        return stEthDecimals;
+    }
+
+    // returns Q112-encoded value
+    function assetToUsd(address bearing, uint amount) public override view returns (uint) {
+        if (amount == 0) return 0;
+        uint _qtyStEthByWstEth = IWstEthToken(bearing).getStETHByWstETH(amount);
+        (uint _currentPriceStEth, bool _isSafePrice) = IStEthPriceFeed(stEthPriceFeed).current_price();
```

Comment:
I suggest using `full_price_info` (Returns the current pool price, whether the price is safe, and the anchor price) and deciding if the price difference is safe by ourselves since malicious admin may set `self.max_safe_price_difference` too low resulting in permanently unsafe prices and blocked liquidations.

Reply:
ok
change to  IStEthPriceFeed(stEthPriceFeed).full_price_info();

`@view
@external
def full_price_info() -> (uint256, bool, uint256):
    """
    @dev Returns the current pool price, whether the price is safe, and the anchor price.
    """
    current_price: uint256 = 0
    has_changed_unsafely: bool = True
    oracle_price: uint256 = 0
    current_price, has_changed_unsafely, oracle_price = self._current_price()
    is_safe: bool = current_price <= 10**18 and not has_changed_unsafely
    return (current_price, is_safe, oracle_price)
`
and  check about ` is_safe: bool = current_price <= 10**18 and not has_changed_unsafely` make by ourselves